extensions/websockets-next/server/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketEndpointBuildItem.java

```diff
@@ -0,0 +1,153 @@
+package io.quarkus.websockets.next.deployment;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.PrimitiveType;
+import org.jboss.jandex.Type;
+import org.jboss.jandex.Type.Kind;
+
+import io.quarkus.arc.processor.BeanInfo;
+import io.quarkus.builder.item.MultiBuildItem;
+import io.quarkus.websockets.next.WebSocket;
+import io.quarkus.websockets.next.runtime.WebSocketEndpoint;
+import io.quarkus.websockets.next.runtime.WebSocketEndpoint.ExecutionModel;
+import io.quarkus.websockets.next.runtime.WebSocketEndpoint.MessageType;
+
+/**
+ * This build item represents a WebSocket endpoint class.
+ */
+public final class WebSocketEndpointBuildItem extends MultiBuildItem {
```

Comment:
Not that it's really important, but with us now having Java 17 as a baseline, these kinds of classes can now be records, no?

Reply:
I don't think that build items should/could be records...

Reply:
Why not?

Reply:
Build Items indeed can't but the `Callback` class can for example.

Reply:
It could but I don't see any benefit here. `Callback` does not define any getters/setters and even the constructor contains some logic...